lib/events/simplestateevents.h

```diff
@@ -46,46 +48,130 @@ namespace EventContent {
     protected:
         QString key;
     };
+
+    template <typename T1, typename T2>
+    class SimpleDualContent {
+    public:
+        using first_value_type = T1;
+        using second_value_type = T2;
+
+        template <typename TT1, typename TT2>
+        SimpleDualContent(QString Key1Name, TT1&& value1, QString Key2Name,
+                          TT2&& value2)
+            : value1(std::forward<TT1>(value1))
+            , value2(std::forward<TT2>(value2))
+            , key1(std::move(Key1Name))
+            , key2(std::move(Key2Name))
+        { }
+
+        SimpleDualContent(const QJsonObject& json, QString key1Name,
+                          QString key2Name)
+            : value1(fromJson<T1>(json[key1Name]))
+            , value2(fromJson<T2>(json[key2Name]))
+            , key1(std::move(key1Name))
+            , key2(std::move(key2Name))
+        { }
+
+        QJsonObject toJson() const
+        {
+            return { { key1, Quotient::toJson(value1) },
+                     { key2, Quotient::toJson(value2) } };
+        }
+
+    public:
+        T1 value1;
+        T2 value2;
+
+    protected:
+        QString key1;
+        QString key2;
+    };
```

Comment:
I'm not quite sure it's the best approach here - there are pretty few (standard) event types with exactly two items of content, and in such case I'd probably make a more basic non-template structure for the content. In any case, this doesn't belong to this header file anymore - please separate it (and `RoomCanonicalAliasEvent` definition) into another file.